workspaces/tests/parallel_transactions.rs

```diff
@@ -38,3 +40,42 @@ async fn test_parallel() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_parallel_async() -> anyhow::Result<()> {
+    let worker = workspaces::sandbox().await?;
+    let contract = worker.dev_deploy(STATUS_MSG_CONTRACT).await?;
+    let account = worker.dev_create_account().await?;
+
+    // Create a queue statuses we can check the status of later.
+    let mut statuses = VecDeque::new();
+    for msg in ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"] {
+        let status = account
+            .call(contract.id(), "set_status")
+            .args_json(json!({
+                "message": msg,
+            }))
+            .transact_async()
+            .await?;
+        statuses.push_back(status);
+    }
+
+    // Retry checking the statuses of all transactions until the queue is empty
+    // with all transactions completed.
+    while let Some(status) = statuses.pop_front() {
+        if let Err(_err) = status.status().await {
+            statuses.push_back(status);
+        }
+    }
+
+    // Check the final set message. This should be "j" due to the ordering of the queue.
+    let final_set_msg = account
+        .call(contract.id(), "get_status")
+        .args_json(json!({ "account_id": account.id() }))
+        .view()
+        .await?
+        .json::<String>()?;
```

Comment:
This test doesn't check that all the transactions executed, just that the last one did. Maybe worth adding in a check to check the nonce increase or the count of successful receipts?

Reply:
good idea and added